powermon/libs/apicoordinator.py

```diff
@@ -48,29 +51,44 @@ def __init__(self, adhocTopic : str, announceTopic: str, enabled: bool, refreshI
             return
 
         self.announce_device()
-        mqtt_broker.subscribe(self.adhocTopic, self.adhocCallback)  # QUESTION: why subscribe here?
+        mqtt_broker.subscribe(self.get_addcommand_topic(), self.addcommand_callback)  # QUESTION: why subscribe here?
 
         # mqtt_broker.publish(self.announceTopic, self.schedule.getScheduleConfigAsJSON())
 
-    def adhocCallback(self, client, userdata, msg):
+    def get_addcommand_topic(self):
+        return self.adhoc_topic_format.format(device_id=self.device.identifier)
+
+    def addcommand_callback(self, client, userdata, msg):
         log.info(f"Received `{msg.payload}` on topic `{msg.topic}`")
-        yamlString = msg.payload.decode("utf-8")
-        log.debug(f"Yaml string: {yamlString}")
-        try:
-            _command_config = yaml.safe_load(yamlString)
-        except yaml.YAMLError as exc:
-            log.error(f"Error processing config file: {exc}")
-
-        for command in _command_config["commands"]:
-            log.debug(f"command: {command}")
-            log.debug(f"self: {self}")
-            # self.schedule.addOneTimeCommandFromConfig(command)
+        jsonString = msg.payload.decode("utf-8")
+        log.debug(f"Yaml string: {jsonString}")
+
+        dto = CommandDTO.parse_raw(jsonString)
+        
+        trigger = Trigger.from_DTO(dto.trigger)    
+        command = Command.from_DTO(dto)
+        Command(code=dto.command_code, commandtype="basic", outputs=[], trigger=trigger)
+        outputs = []
+        
+        
+        output = API_MQTT(formatter=SimpleFormat({}))
+        outputs.append(output)
+            
+        command.set_outputs(outputs=outputs)
+        command.set_mqtt_broker(self.mqtt_broker)
+        
+        self.device.add_command(command)
+
+        #self.announce_device()
+        
+        return command
+            
 
     def run(self):
         # QUESTION: do we need a run?
         if not self.enabled:
             return
-        if not self.last_run or time() - self.last_run > self.refreshInterval:
+        if not self.last_run or time() - self.last_run > self.refresh_interval:
             log.info("APICoordinator running")
             self.announce_device()  # QUESTION: what are we announcing here?
```

Comment:
This is putting the information about the device on the topic "powermon/announce". This gets read by the API then it subscribes to the "powermon/{device_id}/results/#" topic to get results from commands being executed. I haven't tested it yet but this should allow the API to work with multiple devices sending results.